### PR TITLE
[boost] Don't clone unused libs

### PR DESCRIPTION
**Pull request by AndrewShkrob**
-90 unused libs from boost

How to update boost:
* Update `3party/boost`
* Run `./configure.sh` or `./configure_boost.sh`

How to add/remove boost lib:
* Modify `configure_boost.sh`
* Run `./configure.sh` or `./configure_boost.sh`

Really simple and should save some space and time 😊

TODO:
- [ ] Add arguments to /configure_boost.sh to allow separate execution of checkout and configuration tasks in the appropriate CI jobs.

### CODE REVIEW

**Comment by vng** [comment]
Well, if you decided to play here, why not remove boost from 3party at all? Everybody will install the boost within their favorite package manager.

**Comment by vng** [comment]
I have 3 more projects on my machine and all of them use boost from /opt/homebrew/..

**Comment by biodranik** [comment]
This solution complicates the future support and may cause build errors if some dependencies change internally. Which problem is solved by this PR? Faster checkout time should be solved in a different way, for example, by storing a copy of boost in the repo (or downloading and unzipping the boost archive locally when running configure.sh).

File: configure_boost.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  libs/utility
+  libs/uuid
+  libs/variant
+  tools
+)
+
+git submodule update --depth 1 --init --recursive --jobs=$JOBS_COUNT ${BOOST_SUBMODULES[@]}
+
+if [[ "$OSTYPE" == msys ]]; then
+  echo "For Windows please run:"
```
Comment (rtsisyk) [inline configure_boost.sh:83]: Why don't just run it instead of printing instructions?

File: configure.sh [hunk 2]
```diff
@@ -131,27 +131,4 @@ else
...
-  echo "cd 3party\\boost"
-  echo "bootstrap.bat"
-  echo "b2 headers"
-  echo "cd ..\\.."
-else
-  ./bootstrap.sh
-  ./b2 headers
-fi
-cd "$BASE_PATH"
+./configure_boost.sh
```
Comment (rtsisyk) [inline configure.sh:134]: Just keep it in the same file, why not?

**Comment by rtsisyk** [comment]
+1 for removing boost from thirdparty.

File: configure_boost.sh (hunk 1 above)
Comment (biodranik) [reply configure_boost.sh:83]: It should be done from cmd shell.

**Comment by AndrewShkrob** [comment]
I'm closing this PR because nobody liked this idea

**Pull request closed without merging**